src/lib.rs

```diff
@@ -103,16 +103,19 @@ impl Build {
             .arg("no-comp")
             .arg("no-zlib")
             .arg("no-zlib-dynamic")
+            // MUSL doesn't implement some of the libc functions that the async
+            // stuff depends on, and we don't bind to any of that in any case.
+            .arg("no-async");
+
+        if !target.contains("x86_64-unknown-linux-gnu") {
```

Comment:
I think better check for `musl` specifically so that other OS can also utilize engine.